lib/parser/open_element_stack.js

```diff
@@ -278,12 +277,13 @@ OpenElementStack.prototype.hasInScope = function (tagName) {
 
 OpenElementStack.prototype.hasNumberedHeaderInScope = function () {
     for (var i = this.stackTop; i >= 0; i--) {
-        var tn = this.treeAdapter.getTagName(this.items[i]);
+        var tn = this.treeAdapter.getTagName(this.items[i]),
+            ns = this.treeAdapter.getNamespaceURI(this.items[i]);
 
-        if (tn === $.H1 || tn === $.H2 || tn === $.H3 || tn === $.H4 || tn === $.H5 || tn === $.H6)
+        if ((tn === $.H1 || tn === $.H2 || tn === $.H3 || tn === $.H4 || tn === $.H5 || tn === $.H6) && ns === NS.HTML)
```

Comment:
IIUC, those functions are pretty cold, and any tree adapter implementation are assumed to be fast, too, but otherwise I would probably split this into two `if`s where external one calls only tag name, and if it matches, only then calls `getNamespaceURI` to check it too, thus creating a "fast-case" for any elements in stack that are not headers.

But, as I said, might be not beneficial in this particular case, dunno.